build-system/tasks/runtime-test.js

```diff
@@ -541,96 +549,141 @@ function runTests() {
   const sectionMarker =
       (argv.saucelabs || argv.saucelabs_lite) ? 'saucelabs' : 'local';
 
-  let resolver;
-  const deferred = new Promise(resolverIn => {resolver = resolverIn;});
-  new Karma(c, function(exitCode) {
-    if (shouldCollapseSummary) {
-      console./* OK*/log('travis_fold:end:console_errors_' + sectionMarker);
-    }
-    server.emit('kill');
-    if (exitCode) {
-      log(
-          red('ERROR:'),
-          yellow('Karma test failed with exit code ' + exitCode));
+  // Run Sauce Labs tests in batches to avoid timeouts when connecting to the
+  // Sauce Labs environment.
+  let processExitCode;
+  if (argv.saucelabs || argv.saucelabs_lite) {
+    processExitCode = await runTestInBatches();
+  } else {
+    processExitCode = await createKarmaServer(c);
+  }
+
+  server.emit('kill');
+  exitCtrlcHandler(handlerProcess);
+
+  if (processExitCode != 0) {
+    log(
+        red('ERROR:'),
+        yellow('Karma test failed with exit code ' + processExitCode));
+    process.exitCode = 1;
+  }
+
+  /**
+   * Runs tests in batches
+   * @return {string} processExitCode
+   */
+  async function runTestInBatches() {
+    let batch = 1;
+    let startIndex = 0;
+    let endIndex = batchSize;
+    let processExitCode = 0;
+
+    log(green('Testing on ' + saucelabsBrowsers.length +
+      ' Sauce Lab browsers in total'));
```

Comment:
Also, you can use the backtick operator to avoid string concatenation:
```
log(green(`Testing on ${saucelabsBrowsers.length` Sauce Lab browser(s) in total`));
```

Reply:
@rsimha Updated the message.
@danielrozenberg Using backtip would be nice but it purposely includes line breaks which we don't want.